src/runtime/assemblymanager.cs

```diff
@@ -284,6 +287,17 @@ internal static void ScanAssembly(Assembly assembly)
                 if (ns != null)
                 {
                     namespaces[ns].TryAdd(assembly, string.Empty);
+                    try
+                    {
+                        namespaceAdded?.Invoke(ns);
+                    }
+                    catch (Exception e)
+                    {
+                        // For some reason, exceptions happening here does... nothing.
+                        // Even System.AccessViolationExceptions gets ignored.
+                        Console.WriteLine($"Namespace added callback failed with: {e}");
+                        throw;
```

Comment:
Why catch and rethrow? Just to get a log?

Reply:
The throw does nothing; the caller catches *all* exceptions. It might be a bug in the runtime (it might throw appropriately in mono), it might change in the future, so I'm rethrowing. I'm also logging because it took a long time to find why thing were borken further down the execution.